src/RequestHeaderParser.php

```diff
@@ -15,15 +15,20 @@
 class RequestHeaderParser extends EventEmitter
 {
     private $buffer = '';
-    private $maxSize = 4096;
+    private $maxSize;
 
     private $localSocketUri;
     private $remoteSocketUri;
 
-    public function __construct($localSocketUri = null, $remoteSocketUri = null)
+    public function __construct($localSocketUri = null, $remoteSocketUri = null, $maxSize = 4096)
     {
+        if (!is_integer($maxSize)) {
+            throw new \InvalidArgumentException('Invalid type for maxSize provided. Expected an integer value.');
+        }
```

Comment:
Not sure how much value this check provides? I would vote to remove this and replace this with a docblock (which can be replaced by PHP 7+ typehints eventually).

Reply:
@clue how we want to continue here?